### PR TITLE
Adds new virus effects, adds virus say handling

### LINKED ISSUE #12511: Pathogenic Incubator Bug
(WEB REPORT BY: riekland REMOTE: 172.93.109.202:7777)
>Revision
66e2d4d61d281895120a9b9aa9f127d5be3f103d 

>General description of the issue
When upgraded, the pathogenic incubator will incubate forever if told to isolate a virus

>What you expected to happen
for it to isolate it faster than before

>What actually happened
it never isolated

>Steps to reproduce if possible
Upgrade the pathogenic incubator and try to isolate a virus from a blood sample

### PR DESCRIPTION
**Pull request by ghost**
About 40% tested, 110% evil

The say manipulation works largely in the same way that mask chat changes do. I've added 2 viruses that use this new say manipulation so as to demonstrate how to make more in the future

While working with viruses, I found that the var/multiplier would never be actually given a value, so now it should, so now virology has a little more autism to work with

fixes #12511 

:cl:

- rscadd: Adds say manipulation to viruses, so they can now affect what people are saying
- rscadd: "Adds Hangman Syndrome, G__d l_ck"
- rscadd: "Adds Self-lubricating Footstep Syndrome, it's as interesting as it sounds"
- rscadd: Adds Gyroscopic Manipulation Syndrome
- rscadd: "Adds Pro-tagonista syndrome"
- tweak: "The previously useless 'Multiplier' for virus effects now actually works"
- tweak: Virology machines are now fixed, and in some places buffed. It is now safe to upgrade them.

### CODE REVIEW

File: code/modules/virus2/effect.dm [hunk 1]
```diff
@@ -928,3 +931,200 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
...
+				spawn(800)
+					if (istype(T) && T.wet >= 2)
+						T.wet = 0
+						if(T.wet_overlay)
+							T.overlays -= T.wet_overlay
+							T.wet_overlay = null
+
+/obj/item/clothing/shoes/clown_shoes/slippy/dropped(mob/user as mob)
+	canremove = 1
+	..()
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm]: You should put these three blocks in the file with the clown shoes.

File: code/modules/virus2/effect.dm [hunk 2]
```diff
@@ -928,3 +931,200 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
 
 		to_chat(mob, "Suddenly, your knowledge of languages comes back to you.")
 	..()
+
+
+/datum/disease2/effect/lubefoot
+	name = "Self-lubricating Footstep Syndrome"
+	stage = 3
+
+/datum/disease2/effect/lubefoot/activate(var/mob/living/carbon/mob,var/multiplier)
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm:904]: You forgot a deactivate to remove the shoes or make them removable or something.

File: code/modules/virus2/effect.dm [hunk 3]
```diff
@@ -928,3 +931,200 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
...
+/datum/disease2/effect/anime_hair/affect_mob_voice(var/datum/speech/speech)
+	var/message=speech.message
+
+	if(prob(20))
+		message += pick(" Nyaa", "  nya", " , Nyaa~", "~")
+
+	speech.message = message
+
+/obj/item/clothing/head/kitty/cursed
+	canremove = 0
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm]: Move in with the kitty ears

File: code/modules/virus2/effect.dm [hunk 4]
```diff
@@ -928,3 +931,200 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
...
+					//REAL katana /obj/item/weapon/katana
+					var/obj/item/weapon/katana/real_katana = new /obj/item/weapon/katana
+					affected.put_in_hands(real_katana)
+				else
+					//Toy katana /obj/item/toy/katana
+					var/obj/item/toy/katana/fake_katana = new /obj/item/toy/katana
+					affected.put_in_hands(fake_katana)
+				given_katana = 1
+
+datum/disease2/effect/anime_hair/deactivate(var/mob/living/carbon/mob)
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm:1056]: Kitty ears meant to stay cursed forever?

**Review by Intigracy — CHANGES_REQUESTED** [review]

File: code/modules/virus2/effect.dm [hunk 5]
```diff
@@ -928,3 +931,200 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
...
+	message = replacetext(message, "o", "_")
+	message = replacetext(message, "O", "_")
+	message = replacetext(message, "u", "_")
+	message = replacetext(message, "U", "_")
+
+	speech.message = message
+
+
+/datum/disease2/effect/anime_hair
+	name = "Pro-tagonista Syndrome"
```
Comment (JustSumBody) [inline code/modules/virus2/effect.dm:968]: You are not the protagonist.

File: code/modules/virus2/effect.dm (hunk 4 above)
Comment (ghost) [reply code/modules/virus2/effect.dm:1056]: Yes

File: code/modules/virus2/effect.dm (hunk 3 above)
Comment (ghost) [reply code/modules/virus2/effect.dm]: Will do. What about the cursed clown mask a bit further up?

File: code/modules/virus2/effect.dm (hunk 1 above)
Comment (ghost) [reply code/modules/virus2/effect.dm]: Will do. What about the cursed clown mask a bit further up?

File: code/modules/virus2/effect.dm (hunk 5 above)
Comment (ghost) [reply code/modules/virus2/effect.dm:968]: Remobe vay

File: code/modules/virus2/effect.dm (hunk 2 above)
Comment (ghost) [reply code/modules/virus2/effect.dm:904]: the shoes are meant to be a holdover. Only  way to disable the honk is to remove its feet or acid them

**Comment by ghost** [comment]
All complaints done, likely there's new stuff to complain about!

File: code/modules/clothing/masks/gasmask.dm [hunk 6]
```diff
@@ -170,14 +170,19 @@
...
-/obj/item/clothing/mask/gas/virusclown_hat //why isn't this just a subtype of clown_hat???????
-	name = "clown wig and mask"
-	desc = "A true prankster's facial attire. A clown is incomplete without his wig and mask."
-	icon_state = "clown"
-	item_state = "clown_hat"
-	species_fit = list(VOX_SHAPED)
-	can_flip = 0
-	canstage = 0
+/obj/item/clothing/mask/gas/clown_hat/virus //why isn't this just a subtype of clown_hat??????? //Is now
+	desc = "A true prankster's facial attire. A clown is incomplete without his wig and mask.<span class = 'notice'>On second look, it looks like it's coming out of the wearers skin!</span>"
```
Comment (Intigracy) [inline code/modules/clothing/masks/gasmask.dm]: space after the period

File: code/modules/mob/living/carbon/human/say.dm [hunk 7]
```diff
@@ -46,7 +46,16 @@
 					temp_message[H] = "HONK"
 					pick_list -= H //Make sure that you dont HONK the same word twice
 				speech.message = jointext(temp_message, " ")
-
+	if(virus2.len)
+		//to_chat(world, "Checking for virus")
+		for(var/ID in virus2)
+			var/datum/disease2/disease/V = virus2[ID]
+			//to_chat(world, "Checking for voice effect")
```
Comment (Intigracy) [inline code/modules/mob/living/carbon/human/say.dm]: Go ahead and just remove these commented out debug messages.

File: code/modules/virus2/effect.dm [hunk 8]
```diff
@@ -928,3 +893,184 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
...
+					var/obj/item/weapon/katana/real_katana = new /obj/item/weapon/katana
+					affected.put_in_hands(real_katana)
+				else
+					//Toy katana /obj/item/toy/katana
+					var/obj/item/toy/katana/fake_katana = new /obj/item/toy/katana
+					affected.put_in_hands(fake_katana)
+				given_katana = 1
+
+datum/disease2/effect/anime_hair/deactivate(var/mob/living/carbon/mob)
+	to_chat(mob, "<span class = 'notice'>You no longer feel quite like the main character</notice>")
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm]: punctuation

File: code/modules/virus2/effect.dm [hunk 9]
```diff
@@ -928,3 +893,184 @@ var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/mon
...
+	if(affected.shoes && istype(affected.shoes, /obj/item/clothing/shoes/kneesocks))
+		affected.shoes.canremove = 1
+	if(affected.w_uniform && istype(affected.w_uniform, /obj/item/clothing/under/schoolgirl))
+		affected.w_uniform.canremove = 1
+
+/datum/disease2/effect/anime_hair/affect_mob_voice(var/datum/speech/speech)
+	var/message=speech.message
+
+	if(prob(20))
+		message += pick(" Nyaa", "  nya", " , Nyaa~", "~")
```
Comment (Intigracy) [inline code/modules/virus2/effect.dm]: " , Nyaa~"
what?

**Review by Intigracy — CHANGES_REQUESTED** [review]
should be it

File: code/modules/virus2/effect.dm (hunk 9 above)
Comment (ghost) [reply code/modules/virus2/effect.dm]: This is one of those questions where I appear autistic when explaining it, so I shan't bother

File: code/modules/virus2/effect.dm (hunk 8 above)
Comment (ghost) [reply code/modules/virus2/effect.dm]: Specification

Comment (Intigracy) [reply code/modules/virus2/effect.dm]: The end of the line.

File: code/modules/virus2/effect.dm (hunk 9 above)
Comment (Intigracy) [reply code/modules/virus2/effect.dm]: message = "Help I'm being killed in the dorms!"
message += " , Nyaa~"
message = "Help I'm being killed in the dorms! , Nyaa~"

**Comment by ghost** [comment]
Don't merge just yet, realized that how I've gated some of the symptoms, it is potentially impossible for it to actually get to them

**Comment by ghost** [comment]
Okay, should be goooood to go!

**Comment by ghost** [comment]
Eep

**Pull request merged**